.devcontainer/Dockerfile

```
@@ -0,0 +1,49 @@
+    JDK_HOME_DIR=$(tar -tf "/tmp/jdk.tar.gz" | awk 'NR==1 {split($0, a, "/"); print a[1]}') && \
+    # Rename the extracted directory to a consistent name
+    mv "${JDK_INSTALL_DIR}/${JDK_HOME_DIR}" "${JDK_INSTALL_DIR}/jdk" && \
+    rm "/tmp/jdk.tar.gz"
+
+FROM node:24 as graphite_installer
+RUN npm install -g @withgraphite/graphite-cli@stable
+
+FROM mcr.microsoft.com/devcontainers/base:ubuntu-24.04 as devcontainer
+
+# Required to compress speech files in the tts.py script
+RUN apt-get update && \
```

Hadolint (2.12.0)

[warning] Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`

(DL3008)